fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -6344,38 +6355,31 @@ public Map<String, Object> undoLastDisbursal(ScheduleGeneratorDTO scheduleGenera
     /**
      * Reverse only disbursement, accruals, and repayments at disbursal transactions
      *
-     * @param actualDisbursementDate
+     * @param lastDisbursalTransaction
      * @return
      */
-    public List<LoanTransaction> reverseExistingTransactionsTillLastDisbursal(LocalDate actualDisbursementDate) {
-        final List<LoanTransaction> reversedTransactions = new ArrayList<>();
+    public void reverseExistingTransactionsTillLastDisbursal(LoanTransaction lastDisbursalTransaction) {
         for (final LoanTransaction transaction : this.loanTransactions) {
-            if ((actualDisbursementDate.equals(transaction.getTransactionDate())
-                    || actualDisbursementDate.isBefore(transaction.getTransactionDate()))
+            if ((transaction.getId().compareTo(lastDisbursalTransaction.getId()) >= 0)
```

Comment:
We need to check the transactionDate as well as it was originally to avoid reverting a backdated transaction!